Add DEVELOPMENT.md

a-roberts opened the pull request:
For https://github.com/tektoncd/dashboard/issues/23, review comments, testing and feedback welcome, it references using Dockerfile_test which is at PR https://github.com/tektoncd/dashboard/pull/21 and should be merged first.

@jessm12 @dibbles please take a look 👼.

dibbles commented on DEVELOPMENT.md:
```diff
@@ -0,0 +1,108 @@
+_Adding the `upstream` remote sets you up nicely for regularly
+[syncing your fork](https://help.github.com/articles/syncing-a-fork/)._
+
+### Requirements
```
Is it worth putting this "Requirements" section ahead of the checking out your fork as the checkout commands assume a GOPATH already exists?

dibbles commented on DEVELOPMENT.md:
```diff
@@ -0,0 +1,108 @@
+Docker for Desktop using an edge version has been proven to work for both
+developing and running Pipelines and the dashboard. Your Kubernetes version must be 1.11 or later.
+
+## Environment Setup
```
Is this section redundant as same information in Iterating section below.

dibbles commented on DEVELOPMENT.md:
```diff
@@ -0,0 +1,108 @@
+
+## Access the dashboard
+
+`kubectl port-forward $(kubectl get pod -l app=tekton-dashboard -o name) 9097:9097`
```
Might need to expand on this as currently this is just the back end?? 50/50 because will just end up taking out any alteration here once the front-end is added ...... although the front-end would likely be ingressed one would guess

dibbles commented on README.md:
```diff
@@ -8,10 +8,18 @@ Tekton Dashboard is a general purpose, web-based UI for Tekton Pipelines. It all
-[coming soon] To deploy the dashboard, execute the following command:
-
+**Currently** to view the backend APIs, at `localhost:9087/v1/namespaces/<namespace>/<resource name>` for example:
+  
```
Lets not forget you currently still need to build/push and alter the yaml before the apply

a-roberts replied:
D'oh, will fix 😄

a-roberts commented:
Think I've addressed all of the comments now, @dibbles can you please take a look again and @skaegi if you're happy too let's get this merged or further worked on? 

Feedback welcome, perhaps we want to mention more about the frontend now? I'm a backend coder rn so likely missed loads - what else should we be adding? Thanks

vdemeester commented on DEVELOPMENT.md:
```diff
@@ -0,0 +1,105 @@
+
+You must install these tools:
+
+1. [`go`](https://golang.org/doc/install): The language Tekton Dashboard is
```
Don't we need `node` too ? (for react & co)

a-roberts replied:
Yep mb, adding the frontend parts next - indeed node and npm for local frontend dev is required 😄

a-roberts replied:
I've discussed this with @AlanGreene who works on our frontend, he's kindly agreed to run with this as a new PR. We should get what we have in for now as this gives us room to add the API definitions for our backend next and we'll have something in place at least that lets people know how to build and run the backend. I'm going to call out that the frontend reqs are coming up!

vdemeester commented on DEVELOPMENT.md:
```diff
@@ -0,0 +1,105 @@
+
+Many of the instructions here replicate what's in the [tektoncd/pipeline development guide](https://github.com/tektoncd/pipeline/blob/master/DEVELOPMENT.md), with a couple of caveats currently.
+
+1. This project currently does not support being built with `ko`
```
Out of curiosity, what doesn't work with `ko` ?

a-roberts replied:
I'll be upfront - we haven't tried it for this project (only for knative repositories and tektoncd/pipeline), I had a look around the pipeline repo and I noticed lots of `kodata` folders so I'm unsure about using it just yet, perhaps we need it? Would be awesome if it all worked!

AlanGreene commented on README.md:
```diff
@@ -8,10 +8,18 @@ Tekton Dashboard is a general purpose, web-based UI for Tekton Pipelines. It all
 
-[coming soon] To deploy the dashboard, execute the following command:
-
+**Currently** to view the backend APIs, at `localhost:9087/v1/namespaces/<namespace>/<resource name>` for example:
```
9097 here to match the command below?

AlanGreene commented:
/lgtm

jessm12 commented on DEVELOPMENT.md:
```diff
@@ -0,0 +1,106 @@
+
+## Environment Setup
+
+We recommend Docker building, pushing, and replacing the yaml in `install` to refer to your built and pushed image.
```
This sentence is confusing to me - replacing the yaml? maybe specify here which field in the yaml to alter

jessm12 commented on DEVELOPMENT.md:
````diff
@@ -0,0 +1,106 @@
+`kubectl config current-context`):
+
+```shell
+kubectl apply -f `install/tekton-dashboard-deployment.yaml``
````
Could this say simply `kubectl apply -f install/`

jessm12 commented on DEVELOPMENT.md:
```diff
@@ -0,0 +1,106 @@
+1. [`dep`](https://github.com/golang/dep): For managing external Go
+   dependencies. - Please Install dep v0.5.0 or greater.
+1. [`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/): For
+   interacting with your kube cluster
```
Maybe we could mention a 'safe' kubectl version here as we know there's a problem with creds creation in some versions

jessm12 commented on DEVELOPMENT.md:
```diff
@@ -0,0 +1,106 @@
+
+1. Docker build and push your image of the dashboard
+1. Update your `install` yaml to refer to your image location
+1. Run the Go tests, for example with: `docker build -f Dockerfile_test .`
```
Is this an example? or is this at the moment how we recommend to run tests

jessm12 commented:
/approve

bobcatfish commented:
:heart_eyes_cat:
